pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -28,13 +30,32 @@ func New(first []byte) *Fingerprint {
 
 func NewFromFile(file *os.File, size int) (*Fingerprint, error) {
 	buf := make([]byte, size)
+	if hasGzipExtension(file.Name()) {
```

Comment:
Shouldn't fingerprints be created from decompressed data only if `compression` is configured to `gzip` or `auto`? Creating fingerprints from decompressed data if the data is sent in compressed form seems wrong.